pino.js

```diff
@@ -160,23 +172,33 @@ Object.defineProperty(pinoPrototype, 'child', {
     }
     data = this.chindings + data.substr(0, data.length - 1)
 
+    let levelChild = bindings.level || this.level
+    const level = this.levelMap
+    Object.keys(level).sort(tools.byPrecision).map((pattern) => {
+      const regexp = new RegExp('^' + pattern.replace(/\*/g, '.*?') + '$')
+      if (bindings.name && regexp.test(bindings.name)) levelChild = level[pattern]
+    })
```

Comment:
You are right, and actually most of the work can be done at the time of setting the `level` property, long before children are created. So now pre-computing the regexps and storing them in an array instead of a map. The array is iterated with a for loop.